File: 3834-merge-adjacent-equal-elements/3834-merge-adjacent-equal-elements.cpp
class Solution {
    using ll = long long;
    using vll = vector<ll>;
public:
    vector<long long> mergeAdjacent(vector<int>& nums) {
        vector<ll> res;
        for (const auto& x : nums) {
            ll cur = x;
            while (!res.empty() && res.back() == cur) {
                cur *= 2;
                res.pop_back();
            }
            res.push_back(cur);
        }
        return res;
    }
};

File: 3834-merge-adjacent-equal-elements/3834-merge-adjacent-equal-elements.py
class Solution:
    def mergeAdjacent(self, nums: List[int]) -> List[int]:
        res = []
        for x in nums:
            curr = x
            while res and res[-1] == curr:
                curr *= 2
                res.pop()
            
            res.append(curr)
        
        return res